### PR TITLE
Molecule: Adds a to_file function

### PR DESCRIPTION
**Pull request by dgasmith**
Simplifies several tests as well.

Closes https://github.com/MolSSI/QCFractal/issues/294.

### CODE REVIEW

File: qcelemental/models/molecule.py [hunk 1]
```diff
@@ -610,6 +614,42 @@ def from_file(cls, filename, dtype=None, *, orient=False, **kwargs):
...
+
+        Parameters
+        ----------
+        filename : str
+            The filename to write to
+        dtype : Optional[str], optional
+            The type of file to write, attempts to infer dtype from the filename if not provided.
+
+        """
+        ext = os.path.splitext(filename)[1]
```
[the remark this answers is not in the document]
Comment (dgasmith) [reply qcelemental/models/molecule.py]: Where does Path come from?

File: qcelemental/models/molecule.py [hunk 2]
```diff
@@ -610,6 +614,42 @@ def from_file(cls, filename, dtype=None, *, orient=False, **kwargs):
...
+
+        if dtype is None:
+            if ext in _extension_map:
+                dtype = _extension_map[ext]
+            else:
+                raise KeyError(f"Could not infer dtype from filename: `{filename}`")
+
+        if dtype in ["xyz", "psi4"]:
+            stringified = self.to_string(dtype)
```
[the remark this answers is not in the document]
Comment (dgasmith) [reply qcelemental/models/molecule.py]: I think so as I would rather we are writing a specification. A user can always work around of course.

File: qcelemental/models/molecule.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (loriab) [reply qcelemental/models/molecule.py]: I was suggesting pathlib syntax

File: qcelemental/models/molecule.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (loriab) [reply qcelemental/models/molecule.py]: ok then

File: qcelemental/models/molecule.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (dgasmith) [reply qcelemental/models/molecule.py]: Gotcha, switched over.

**Pull request merged**